### ISSUE TITLE
Make `_convert_time`  public

### ISSUE THREAD

**Issue by priba** [mosaicml/composer#2426]
## 🚀 Feature Request
Expose `_convert_time` in the API. That is `_convert_time` -> `convert_time`.

https://github.com/mosaicml/composer/blob/915fab73c43af754150929ae03120a67acbe60f1/composer/optim/scheduler.py#L127

## Motivation

For custom callbacks, it would be nice to expose `_convert_time` in the API.

## Additional context

I am happy to submit a PR if this is something that makes sense.

**Comment by mvpatel2000** [comment]
Agree! This would be helpful. If you want to open a PR, that would be great. I think it'd be useful to have a state function that exposes this functionality?

**Comment by priba** [comment]
I thought on moving the function to `core/time.py`. What do you mean by state function?

**Comment by mvpatel2000** [comment]
> I thought on moving the function to core/time.py.

`core/time.py` is only classes, so it might be simpler to just leave definitions there. 

> What do you mean by state function?

Instead, I was thinking of attaching this as a function to `class State` in `core/state.py`. The signature would change to no longer take a state as an argument and just use self instead. What do you think?
